src/main/java/core/basesyntax/dao/StorageDaoImpl.java

```diff
@@ -8,15 +8,22 @@ public class StorageDaoImpl implements StorageDao {
 
     @Override
     public User add(User user) {
-        user.setId(++index);
+        if (user == null) {
```

Comment:
no need to change this class, validation should be in the registration method

Reply:
Is this a requirement/restriction only for this task? Or more like a general rule of thumb?

I was thinking that responsibility of this unit is to put or retrieve a user from Storage, and in perspective this unit could be used later in other service with different kind of validation. So i think, this unit, should at least check the presence of a login field in `User` class object